src/commands/react.ts

```diff
@@ -0,0 +1,168 @@
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  Could not find a message to react to!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return {resultString: 'Could not find target message', result: {success: false}};
+    }
+
+    const targetMessage = fetchedMessage.array()[0];
+
+    const success = await this.reactService.tempReactToMessage(targetMessage, emoji, message);
+
+    if (success === false) {
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  Could not apply the reaction!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return {resultString: 'Could not apply the reaction', result: {success: false}};
+    }
+
+    return {};
+  }
+
+  /**
+   * React to a specific message given by its Id
+   * @param messageId - The Id of the specific message
+   * @param emojiName - The name of the emoji
+   * @param message - The command message
+   */
+  async reactToMessage(messageId: string, emojiName: string, message: Discord.Message): Promise<ICommandResult> {
+    const emoji = this.emojiService.getEmojiByName(emojiName);
+
+    if (!emoji) {
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  Could not find that emoji!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return { resultString: 'Could not find the emoji', result: {success: false}};
+    }
+
+    let initialId = message.channel.id;
+    let channels: Channel[] = [];
+    channels.push(message.channel);
+
+    // Get all guild channels if in a guild chat
+    if (message.channel.type === 'text') {
+      let guild = message.guild;
+      guild.channels.forEach((guildChannel) => {
+        if (guildChannel.type === 'text' && guildChannel.id !== initialId) {
+          channels.push(guildChannel as TextChannel);
+        }
+      });
+    }
+
+    // Loop through channels to try to apply emoji
+    let success = false;
+    while (channels.length > 0) {
+      let channel = channels.shift();
+      let textBasedChannel = channel as TextChannel | DMChannel | GroupDMChannel;
+      if (!textBasedChannel || !textBasedChannel.lastMessageID) { continue; }
+      try {
+        let targetMessage = await textBasedChannel.fetchMessage(messageId);
+
+        let applied = await this.reactService.tempReactToMessage(targetMessage,
+          emoji, message);
+
+        if (applied) {
+          success = true;
+          break;
+        }
+      } catch (ex) {
+        // continue
+      }
+    }
+
+    if (success === false) {
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  Could not find a message to react to!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return {resultString: 'Could not find target message', result: {success: false}};
+    }
+
+    return {};
```

Comment:
Do we want to create a icommandresult class to return from all commands back to the command handler?

Reply:
I think we already have an ICommandResult though we could look at improvements to it in a different PR to keep this smaller